src/components/generics/vacation-test-forms/vacationRequestForm.tsx

```diff
@@ -0,0 +1,217 @@
+    <FormControl
+      variant="standard"
+      sx={{
+        margin: 1,
+        minWidth: 165,
+        marginBottom: 4
+      }}
+    >
+      <InputLabel>{ strings.vacationRequests.vacationType }</InputLabel>
+      <Select
+        name="type"
+        value={ vacationData.type }
+        onChange={ handleVacationTypeChange }
+        label={ strings.vacationRequests.vacationType }
+      >
+        <MenuItem value={ VacationType.VACATION }>
+          { strings.vacationRequests.vacation }
+        </MenuItem>
+        <MenuItem value={ VacationType.UNPAID_TIME_OFF}>
+          { strings.vacationRequests.unpaidTimeOff }
+        </MenuItem>
+        <MenuItem value={ VacationType.SICKNESS}>
+          { strings.vacationRequests.sickness }
+        </MenuItem>
+        <MenuItem value={ VacationType.PERSONAL_DAYS }>
+          { strings.vacationRequests.personalDays }
+        </MenuItem>
+        <MenuItem value={ VacationType.MATERNITY_PATERNITY }>
+          { strings.vacationRequests.maternityPaternityLeave }
+        </MenuItem>
+        <MenuItem value={ VacationType.CHILD_SICKNESS }>
+          { strings.vacationRequests.childSickness }
+        </MenuItem>
+      </Select>
+    </FormControl>
+  );
+
+  /**
+   * Renders spent vacation days
+   */
+  const renderVacationDaysSpent = () => {
+    // Define the date range to compare with holidays
+    const holidaysFi = new Holidays("FI");
+    const startDate = new Date(vacationData.startDate);
+    const endDate = new Date(vacationData.endDate);
+    let day = 0;
+
+    // Iterate over each date in the date range and check if it is a holiday
+    for (let d = new Date(startDate); d <= endDate; d.setDate(d.getDate() + 1)) {
+      if (!holidaysFi.isHoliday(d) && d.getDay() !== 0) {
+        // eslint-disable-next-line no-plusplus
+        day++;
+      }
+    }
+    return day;
+  };
+
+  /**
+   * Renders vacation comment box
+   */
+  const renderVacationCommentBox = () => {
+    return (
+      <>
+        <Typography variant="h4" style={{ fontSize: 13, marginBottom: "1em" }}>
+          { strings.vacationRequests.amountOfChosenVacationDays }
+          { renderVacationDaysSpent() }
+        </Typography>
+        <TextField
+          id="outlined-multiline-flexible"
+          multiline
+          maxRows={5}
+          label={ strings.vacationRequests.leaveAComment }
+          variant="outlined"
+          value={ vacationData.message }
+          onChange={ handleVacationMessageChange }
+          name="message"
+        />
+      </>
+    );
+  };
+
+  useEffect(() => {
+    const newRequest: VacationData = { ...vacationData, days: renderVacationDaysSpent() };
+    setVacationData(newRequest);
+  }, [vacationData.startDate, vacationData.endDate]);
+  
+  /**
+   * Renders vacation apply button
+   */
+  const renderVacationApplyButton = () => (
+    <Button
+      color="secondary"
+      variant="contained"
+      onClick={() => (onClick())}
```

Comment:
no need for the implicit return parentheses here. In this case I think you could use the following syntax, just mmake sure it still works:
onClick={ onClick }

Reply:
It stops working so i am not chancing it.